### PR TITLE
[NET-623] Disable gui server by default for netclient docker

### PR DESCRIPTION
**Pull request by gabrielseibel1**
## Describe your changes
Disable gui server by default for netclient docker, allowing override with a build arg

## Provide testing steps
docker-build it with no args, run the resulting image: prints "[netclient] disabling gui server"
docker-build it with `--build-arg gui_server_disable=""`, run it: doesn't print the message
the message is bound to be echoed before we run `/root/netclient guiServer disable`, but if you want to be thorough you can also query the gui server apis to see if it's on or not.

### CODE REVIEW

**Review by mattkasun — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -19,5 +19,11 @@ COPY --from=builder /app/netclient-app ./netclient
 COPY --from=builder /app/scripts/netclient.sh .
 RUN chmod 0755 netclient && chmod 0755 netclient.sh
 
+# accepts an arg for gui server to be disabled, which is true by default.
+# if you need the guiserver, build the image with this variable set to ""
+# i.e.: `docker build ... --build-arg gui_server_disable=""` to enable it
+ARG gui_server_disable="true"
```
Comment (abhishek9686) [inline Dockerfile:25]: there is no need add this as arg, you can never use GUI for docker

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -19,5 +19,11 @@ COPY --from=builder /app/netclient-app ./netclient
 COPY --from=builder /app/scripts/netclient.sh .
 RUN chmod 0755 netclient && chmod 0755 netclient.sh
 
+# accepts an arg for gui server to be disabled, which is true by default.
+# if you need the guiserver, build the image with this variable set to ""
+# i.e.: `docker build ... --build-arg gui_server_disable=""` to enable it
```
Comment (abhishek9686) [inline Dockerfile:24]: `gui_server_disable` should accept true or false

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Comment by abhishek9686** [comment]
@gabrielseibel1 resolve merge conflicts

**Comment by Aceix** [comment]
merge confilcts

File: Dockerfile
Comment (abhishek9686) [inline Dockerfile:1]: remove the build arg, are you expecting user to build the image in order enable GUI?
it should just accept env `GUI_SERVER_ENABLED`

File: scripts/netclient.sh
Comment (abhishek9686) [inline scripts/netclient.sh:1]: `/root/netclient guiServer disable` by default 
only check if ${GUI_SERVER_ENABLED} is set to true to enable it

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]

**Pull request merged**